### PR TITLE
ROX-13065: Adding skeleton of DeploymentDetails component

### PR DESCRIPTION
**Pull request by sachaudh**
## Description

This PR adds a DeploymentSideBar and DeploymentDetails component that you can see by clicking on a node. This is just a skeleton and the remaining work was divided into subtasks you can view here: https://issues.redhat.com/browse/ROX-12903#view-subtasks. 

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

## Testing Performed

Screen Shot 2022-10-11 at 3 04 15 PM
Screen Shot 2022-10-11 at 3 04 18 PM

### CODE REVIEW

File: ui/apps/platform/src/Containers/NetworkGraph/NetworkGraph.tsx [hunk 1]
```diff
@@ -99,17 +107,25 @@ const TopologyComponent = () => {
...
-                <TopologySideBar show={!!selectedId} resizable onClose={() => setSelectedId(null)}>
-                    <div style={{ height: '100%' }}>{selectedId}</div>
+                <TopologySideBar
+                    show={!!selectedObj}
+                    resizable
+                    onClose={() => setSelectedObj(null)}
+                >
+                    {selectedObj?.type === 'group' && <div>Group</div>}
+                    {selectedObj?.type === 'node' && <DeploymentSideBar />}
+                    {selectedObj?.type === 'edge' && <div>Edge</div>}
```
Comment (alwayshooin) [inline ui/apps/platform/src/Containers/NetworkGraph/NetworkGraph.tsx:122]: i don't think we're going to have a separate side panel for edges from what i recall. @alanonthegit correct me if i'm wrong

File: ui/apps/platform/src/Containers/NetworkGraph/NetworkGraph.tsx [hunk 2]
```diff
@@ -99,17 +107,25 @@ const TopologyComponent = () => {
...
                 </TopologySideBar>
             }
-            sideBarOpen={!!selectedId}
+            sideBarOpen={!!selectedObj}
```
Comment (alwayshooin) [inline ui/apps/platform/src/Containers/NetworkGraph/NetworkGraph.tsx:125]: you can have the side bar be open but not shown? that seems a little confusing to me. is that case for like a minimized side panel?

**Review by alwayshooin — APPROVED** [review]
looks great! just a few questions in the code for you

File: ui/apps/platform/src/Containers/NetworkGraph/NetworkGraph.tsx (hunk 2 above)
Comment (sachaudh) [reply ui/apps/platform/src/Containers/NetworkGraph/NetworkGraph.tsx:125]: Do you mean how TopologyView has the sideBarOpen prop and the TopologySideBar has the show prop? I tested it out and seems like we need sideBarOpen but the show prop didn't seem necessary. I removed that one

File: ui/apps/platform/src/Containers/NetworkGraph/NetworkGraph.tsx (hunk 1 above)
Comment (sachaudh) [reply ui/apps/platform/src/Containers/NetworkGraph/NetworkGraph.tsx:122]: Yea true. Let me remove that

**Pull request merged**